src/lib/documentAnalysis/spellingGrammar/parallelSpellingGrammarWorkflow.ts

```diff
@@ -0,0 +1,259 @@
+      duration: number;
+    }>[] = [];
+    
+    // Process chunks in batches
+    for (let i = 0; i < chunks.length; i += maxConcurrency) {
+      const batch = chunks.slice(i, Math.min(i + maxConcurrency, chunks.length));
+      const batchStartTime = Date.now();
+      
+      const batchPromises = batch.map(async (chunk, batchIndex) => {
+        const chunkIndex = i + batchIndex;
+        const chunkStartTime = Date.now();
+        
+        // Add a small staggered delay to avoid overwhelming the API
+        const staggerDelay = batchIndex * 500; // 500ms stagger between concurrent requests
+        if (staggerDelay > 0) {
+          await new Promise(resolve => setTimeout(resolve, staggerDelay));
+        }
+        
+        logger.info(`Analyzing chunk ${chunkIndex + 1}/${chunks.length} (lines ${chunk.startLineNumber}-${chunk.startLineNumber + chunk.lines.length - 1})`);
+        
+        try {
+          const result = await analyzeChunk(chunk, {
+            agentName: agentInfo.name,
+            primaryInstructions: agentInfo.primaryInstructions || "Find all spelling, grammar, punctuation, and capitalization errors."
+          });
+          
+          const duration = Date.now() - chunkStartTime;
+          logger.info(`Found ${result.highlights.length} errors in chunk ${chunkIndex + 1} (${duration}ms)`);
+          
+          return {
+            chunkIndex,
+            chunk,
+            highlights: result.highlights,
+            duration
+          };
+        } catch (error) {
+          logger.error(`Error analyzing chunk ${chunkIndex + 1}:`, error);
+          return {
+            chunkIndex,
+            chunk,
+            highlights: [],
+            duration: Date.now() - chunkStartTime
+          };
+        }
+      });
+      
+      // Wait for this batch to complete before starting the next
+      const batchResults = await Promise.all(batchPromises);
+      chunkPromises.push(...batchResults.map(r => Promise.resolve(r)));
+      
+      logger.info(`Completed batch ${Math.floor(i / maxConcurrency) + 1}/${Math.ceil(chunks.length / maxConcurrency)} in ${Date.now() - batchStartTime}ms`);
+    }
+    
+    // Wait for all chunks to complete
+    const chunkResults = await Promise.all(chunkPromises);
+    
+    // Sort results by chunk index to maintain order
+    chunkResults.sort((a, b) => a.chunkIndex - b.chunkIndex);
+    
+    // Aggregate highlights and create tasks
+    const allHighlights: SpellingGrammarHighlight[] = [];
+    for (const result of chunkResults) {
+      allHighlights.push(...result.highlights);
+      
+      tasks.push({
+        name: `Analyze chunk ${result.chunkIndex + 1}`,
+        modelName: "claude-3-haiku", // Default model for parallel processing
+        priceInDollars: 0, // Cost tracking handled elsewhere
+        timeInSeconds: result.duration / 1000,
+        log: `Analyzed chunk ${result.chunkIndex + 1}: ${result.highlights.length} errors found`,
```

Comment:
This should be more detailed. The log is cheap to store. It should provide the chunk length and list the errors. Maybe give the opening 200chars or so of the chunk